lib/strtoofft.c

```diff
@@ -129,19 +128,15 @@ static curl_off_t strtooff(const char *nptr, char **endptr, int base)
   }
 
   /* Loop handling digits. */
-  value = 0;
-  overflow = 0;
   for(i = get_char(end[0], base);
       i != -1;
       end++, i = get_char(end[0], base)) {
-    newval = base * value + i;
-    if(newval < value) {
-      /* We've overflowed. */
-      overflow = 1;
+
+    if(value > CURL_OFF_T_MAX / base - i) {
```

Comment:
I believe the correct test is the following otherwise you compare ``(value + i) * base > CURL_OFF_T_MAX`` instead of ``value * base + i > CURL_OFF_T_MAX``  :
```suggestion
    if(value > (CURL_OFF_T_MAX - i) / base) {
```

Reply:
Ah, yes thanks.

I also suspect we have no CI build that actually use this code...